hw4_numpy/numpy_challenge.py

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[ ]:
+
+
+import numpy as np
+
+if __name__ == "__main__":
+        ar1 = np.array([1, 2, 21])
+        ar2 = np.arange(1,19).reshape(3,-1)
+        ar3 = np.zeros(5)
+
+def matrix_multiplication(h,i):
+    #s = np.matmul(h, i)
+    s = np.dot(h,i)
+    return s
+
+def multiplication_check(L): #L - list of matrix
+    c = []
+    for i in range(len(L)-1):
+        if (L[i].shape[1] == L[i+1].shape[0]):
+            c.append(True)
+        else:
+            c.append(False)
+            break
+    if False not in c:
+        return True
+    else:
+        return False
```

Comment:
1. Снова нейминг, впрочем и дальше тоже
2. Очень много лишних действий
```suggestion
def multiplication_check(matrices_list): #L - list of matrix
    for i in range(len(matrices_list) - 1):
        if matrices_list[i].shape[1] != matrices_list[i + 1].shape[0]:
            return False
    return True
```